realm/realm-library/src/main/java/io/realm/OrderedCollectionChangeSet.java

```diff
@@ -28,6 +30,64 @@
  * change, or an array of {@link Range}s.
  */
 public interface OrderedCollectionChangeSet {
+
+    /**
+     * State describing the nature of the changeset.
+     */
+    public enum State {
```

Comment:
Some of the descriptions of the states apply mainly for Sync use case. It might be confusing to read/introduce this to non Sync usage ...